crates/matrix-sdk/src/room/privacy_settings.rs

```diff
@@ -0,0 +1,440 @@
+use ruma::{
+    api::client::{directory::set_room_visibility, room::Visibility, state::send_state_event},
+    assign,
+    events::{
+        room::{
+            canonical_alias::RoomCanonicalAliasEventContent,
+            history_visibility::{HistoryVisibility, RoomHistoryVisibilityEventContent},
+            join_rules::{JoinRule, RoomJoinRulesEventContent},
+        },
+        EmptyStateKey,
+    },
+    OwnedRoomAliasId, RoomAliasId,
+};
+
+use crate::{Error, Result, Room};
+
+impl Room {
+    /// Update the canonical alias of the room.
+    ///
+    /// Note that publishing the alias in the room directory is done separately.
+    pub async fn update_canonical_alias(&self, new_alias: Option<OwnedRoomAliasId>) -> Result<()> {
+        // Create a new alias event combining both the new and previous values
+        let content = assign!(
+            RoomCanonicalAliasEventContent::new(),
+            { alias: new_alias, alt_aliases: self.alt_aliases() }
+        );
+
+        // Send the state event
+        let request = send_state_event::v3::Request::new(
+            self.room_id().to_owned(),
+            &EmptyStateKey,
+            &content,
+        )?;
+        self.client.send(request, None).await?;
+
+        Ok(())
+    }
+
+    /// Update room history visibility for this room.
```

Comment:
The docs on all these methods are a bit terse, it might be nice to explain briefly what the history visibility is and link to the spec for more info.

This applies to the other methods as well, an example here and there wouldn't hurt either.